src/plugins/data_source_management/public/components/direct_query_data_sources_components/direct_query_data_connection/manage_direct_query_data_connections_table.tsx

```diff
@@ -128,50 +130,77 @@ export const ManageDirectQueryDataConnectionsTable = ({
-            }));
-      })
-      .then((finalData) => {
-        setData(
-          featureFlagStatus
+    const fetchOpenSearchConnections = async (): Promise<DataSourceTableItem[]> => {
+      const fetchConnections = featureFlagStatus
+        ? getDataSources(savedObjects.client)
+        : http.get(`${DATACONNECTIONS_BASE}`);
+
+      return fetchConnections
+        .then((response) => {
+          return featureFlagStatus
+            ? fetchDataSourceConnections(
+                response,
+                http,
+                notifications,
+                true,
+                getHideLocalCluster().enabled
+              )
+            : response.map((dataConnection: DirectQueryDatasourceDetails) => ({
+                id: dataConnection.name,
+                title: dataConnection.name,
+                type:
+                  {
+                    S3GLUE: DatasourceTypeToDisplayName.S3GLUE,
+                    PROMETHEUS: DatasourceTypeToDisplayName.PROMETHEUS,
+                  }[dataConnection.connector] || dataConnection.connector,
+                connectionType: dataConnection.connector,
+                description: dataConnection.description,
+              }));
+        })
+        .then((finalData) => {
+          return featureFlagStatus
             ? finalData.filter((item) => item.relatedConnections?.length > 0)
-            : finalData
-        );
-      })
-      .catch(() => {
-        setData([]);
-        notifications.toasts.addDanger(
-          i18n.translate('dataSourcesManagement.directQueryTable.fetchDataSources', {
-            defaultMessage: 'Could not fetch data sources',
-          })
-        );
-      })
-      .finally(() => {
+            : finalData;
+        })
+        .catch(() => {
+          notifications.toasts.addDanger(
+            i18n.translate('dataSourcesManagement.directQueryTable.fetchDataSources', {
+              defaultMessage: 'Could not fetch data sources',
+            })
+          );
+          return [];
+        });
+    };
+
+    const fetchDirectQueryConnections = async (): Promise<DataSourceTableItem[]> => {
+      try {
+        const dataConnectionSavedObjects = await getDataConnections(savedObjects.client);
+        return dataConnectionSavedObjects.map((obj) => ({
+          id: obj.id,
+          title: obj.attributes.connectionId,
+          type: obj.attributes.type,
+        }));
+      } catch (error: any) {
+        return [];
+      }
+    };
+
+    const fetchAllData = async () => {
+      try {
+        const [openSearchConnections, directQueryConnections] = await Promise.all([
+          fetchOpenSearchConnections(),
+          fetchDirectQueryConnections(),
+        ]);
+        setData([...openSearchConnections, ...directQueryConnections]);
+      } catch (error) {
+        // eslint-disable-next-line no-console
+        console.log('Error fetching OpenSearch and Direct Query Connections', error);
```

Comment:
Should we remove this? The user won't benefit from this. Do we want to show the notification toast instead?

Reply:
added notification toast